### ISSUE TITLE
GUIs closing when internet is lost

### ISSUE THREAD

**Issue by acampove** [microsoft/wslg#1247]
### Windows Version

Microsoft Windows [Version 10.0.22631.3527]

### WSL Version

WSL version: 2.0.14.0 Kernel version: 5.15.133.1-1 WSLg version: 1.0.59 MSRDC version: 1.2.4677 Direct3D version: 1.611.1-81528511 DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp Windows version: 10.0.22631.3527

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.15.133.1-microsoft-standard-WSL2 (root@1c602f52c2e4) (gcc (GCC) 11.2.0, GNU ld (GNU Binutils) 2.37) microsoft/WSL#1 SMP Thu Oct 5 21:02:42 UTC 2023

### Distro Version

Ubuntu, Release:        22.04

### Other Software

This happens with all the applications that use GUIs. Terminator, Evince, etc.

### Repro Steps

1. Open a programs that uses a GUI.
2. Open multiple virtual desktops.
3. Place the GUIs in those desktops, one in each.
4. Interrupt the internet connection. e.g by shaking slightly the ethernet cable.
5. All GUIs dissapear.
6. Open another GUI, e.g. by issuing a command in the terminal. All GUIs will reappear in the current virtual desktop.

### Expected Behavior

1. Open a programs that uses a GUI.
2. Open multiple virtual desktops.
3. Place the GUIs in those desktops, one in each.
4. Interrupt the internet connection. e.g by shaking slightly the ethernet cable.
5. Internet goes away but nothing else happens.

### Actual Behavior

1. Open a programs that uses a GUI.
2. Open multiple virtual desktops.
3. Place the GUIs in those desktops, one in each.
4. Interrupt the internet connection. e.g by shaking slightly the ethernet cable.
5. All GUIs dissapear.
6. Open another GUI, e.g. by issuing a command in the terminal. All GUIs will reappear in the current virtual desktop.

```[tasklist]
### Tasks
- [ ] https://github.com/microsoft/wslg/issues/1098
```

**Comment by acampove** [comment]
OK, the problem just happened yesterday and I decided to get done with this:

[WslLogs-2024-05-25_16-05-36.zip](https://github.com/microsoft/wslg/files/15442304/WslLogs-2024-05-25_16-05-36.zip)

**Comment by acampove** [comment]
Ok, there seems to be a solution for this:

https://github.com/microsoft/wslg/issues/1098#issuecomment-2073387788

It's really sad that this issue, been such a big problem, reported by so many people, took around a year to get fixed by a company that is worth tens of billions and has thousands of engineers, in an operating system that is not even free, like linux...

**Comment by kh-abd-kh** [comment]
I had the same problem. I am connected to the internet through
usb modem every time i connect/disconnect my google chrome/jupyter lab
disappeared and WSL crashes.
 But jupyter lab through
jupyter  lab --no-browser 
and opening from Win edge is ok.

################
Then after updating WSL+Kernels, it worked.
#################

I had a special  hand compiled kernel to enable kernel drivers.
I haven't yet tried to compile a new kernel.

**Comment by kh-abd-kh** [comment]
(base) C:\WINDOWS\system32>wsl --version
WSL version: 2.2.4.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.61
MSRDC version: 1.2.5326
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26091.1-240325-1447.ge-release
Windows version: 10.0.22000.160

**Comment by kh-abd-kh** [comment]
one more comment 
I still have since infinity 
WSL crashs when i exceed the RAM

**Comment by acampove** [comment]
It is strange, because 2.2.3 seems to fix this and you have 2.2.4 WSL @kh-abd-kh I also have 2.2.4 and i haven't seen the issue anymore.

**Comment by kh-abd-kh** [comment]
[acampove](https://github.com/acampove)

you are talking about the RAM problem.
It is since infinity working with sage. I can generate big RAM usage . Sometimes just kernel restart.
but more often crashs. When You do big calculation you don't know what to expect.

The Network was solved completely with 2.2.4. No more at all. I am very heavy user  of WSL 8hr+ per day.
